backend/cfg/cfg.ml

```diff
@@ -110,37 +108,13 @@ let replace_successor_labels t ~normal ~exn block ~f =
       | Float_test { lt; eq; gt; uo } ->
           Float_test { lt = f lt; eq = f eq; gt = f gt; uo = f uo }
       | Switch labels -> Switch (Array.map f labels)
-      | Tailcall (Self _) ->
-          (* CR-someday gyorsh: If there is no [Tailcall Self] then we won't
-             affect [t.fun_tailrec_entry_point_label]. Maybe this case should
-             do nothing and [fun_tailrec_entry_point_label] should
-             unilaterally be updated earlier in this function? *)
-          (* CR-someday gyorsh: Move replace_successor_labels back to
-             disconnect_block.ml ?
-
-             Changing t.fun_tailrec_entry_point_label has effect on other
-             blocks, it's not local to the [block] that is passed as argument
-             to [replace_successor_labels].
-
-             Suppose that there are two "Tailcall Self" sites in the
-             function, say blocks L1 and L2 both have Tailcall Self as their
-             terminator. Then, if we call replace_successor_labels on L1 but
-             not on L2 we get an inconsistent CFG, as there is only one
-             tailrec entry point.
-
-             How do we guarantee that all other blocks that refer to
-             t.fun_tailrec_entry_point_label are updated?
-
-             [replace_successor_labels] is only used in eliminate
-             fallthrough, where it is called on all predecessors of a block
-             that is a fallthrough block. If a predecessor block terminates
-             with tailcall self, then its successor is the block at
-             t.fun_tailrec_entry_point_label and the tailrec entry point
-             block has as its predecessors *all* the "tailcall self" blocks. *)
-          t.fun_tailrec_entry_point_label <-
-            f t.fun_tailrec_entry_point_label;
-          block.terminator.desc
-      | Return | Raise _ | Tailcall (Func _) -> block.terminator.desc
+      | Tailcall (Self { destination; label_after; }) ->
+        Tailcall (Self { destination = f destination; label_after = f label_after; })
```

Comment:
Thinking about `label_after` again, it is not correct to update them here in `replace_successor_labels` using `f`, because `label_after` are not successor labels of Tailcall instructions, and `f` is intended to be applied only to the successor labels.  

It is safe to keep `label_after` unchanged during CFG transformations.
These labels are not used as block entry labels anywhere in Linear and in CFG. The labels are created during `selectgen` and not referenced anywhere, until they are printed out in Emit.  In any case, they are gone in 4.12 when Spacetime is removed. Until then, if you want to be on the safe side, it would be enough to check somewhere global and abort if spacetime is enabled `Config.spacetime` when `use_ocamlcfg` is enabled.